string-edit.el

```diff
@@ -82,15 +82,24 @@ This saves you from needing to manually escape characters."
       (indent-region beg end))))
 
 (defun se/find-original ()
-  (if (derived-mode-p 'js2-mode 'js-mode)
-      (se/js-strings-at-point)
-    (se/string-at-point)))
-
-(defun se/guess-at-major-mode ()
-  (save-excursion
+  (cond
+   ((derived-mode-p 'js2-mode 'js-mode)
+    (se/js-strings-at-point))
+   ((eq major-mode 'emacs-lisp-mode)
+    (se/elisp-string-at-point))
+   (t
+    (se/string-at-point))))
```

Comment:
That would work for the checks where you do it with `major-mode`, but not for `derived-mode-p` or some more elaborate checks.

Maybe it would be worth to extract at least this one common case and leave the rest inlined though.